cpp/log/cert.cc

```diff
@@ -43,6 +43,99 @@ static int X509_get_signature_nid(const X509* x) {
 #endif
 
 
+namespace {
+#if defined(OPENSSL_IS_BORINGSSL)
+// BoringSSL doesn't have DSA hooked up so to accept these certs we have
+// to do the work ourselves. Can be called with a non DSA issuer key but will
+// always return false in that case.
+//
+// cert is the certificate that is to be checked
+// issuer_key is the public key of the certificate issuer (should be DSA)
+// If dsa_key_with_params is not null it's a key that can be used to copy
+// the DSA parameters if they are absent from the issuer key.
+// Returns true if the DSA signature was correctly verified or false if
+// it did not or there was any error along the way.
+StatusOr<bool> check_dsa_signature(const X509* cert, EVP_PKEY* issuer_key,
```

Comment:
Can the issuer key be const? Perhaps it can't because the underlying API isn't correctly const?